File: ui/src/app/business/cluster/cluster.ts
import {BaseModel} from '../../shared/class/BaseModel';

export class Cluster extends BaseModel {
    name: string;
    spec: Spec;
    nodeSize: string;
    status: string;
    preStatus: string;
    provider: string;
    projectName: string;
    source: string;
    multiClusterRepository: string;
}

export class Spec {
    dockerStorageDir: string;
    version: string;
    upgradeVersion: string;
    kubeProxyMode: string;
    networkType: string;
    ciliumTunnelMode: string;
    flannelBackend: string;
    calicoIpv4PoolIpip: string;
    architectures: string;
    runtimeType: string;
}

export class ClusterStatus {
    phase: string;
    prePhase: string;
    message: string;
    conditions: Condition[] = [];
}

export class Condition {
    status: string;
    message: string;
    name: string;
}

export class CreateNodeRequest {
    role: string;
    hostName: string;
}


export class ClusterCreateRequest extends BaseModel {
    name: string;
    version: string;
    provider: string;
    architectures: string;
    networkType: string;
    ciliumVersion: string;
    ciliumTunnelMode: string;
    ciliumNativeRoutingCidr: string;
    runtimeType: string;
    dockerStorageDir: string;
    containerdStorageDir: string;
    flannelBackend: string;
    calicoIpv4poolIpip: string;
    kubeProxyMode: string;
    nodeportAddress: string;
    enableDnsCache: string;
    dnsCacheVersion: string;
    kubernetesAudit: string;
    ingressControllerType: string;
    plan: string;
    nodes: CreateNodeRequest[] = [];
    workerAmount: number;
    dockerSubnet: string;
    projectName: string;
    helmVersion: string;
    networkInterface: string;
    supportGpu: string;
    yumOperate: string;
    clusterCidr: string;
    serviceCidr: string;
    maxPodNum: number;
}

export class CLusterImportRequest {
    name: string;
    apiServer: string;
    token: string;
    router: string;
    projectName: string;
}

export class InitClusterResponse {
    message: string;
}


export class ClusterSecret {
    kubernetesToken: string;
}

export class ClusterUpgradeRequest {
    clusterName: string;
    version: string;
}

export class ClusterHealthCheck {
    hooks: ClusterHealthCheckHook[] = [];
    level: string;
}

export class ClusterHealthCheckHook {
    name: string;
    level: string;
    msg: string;
}

export class ClusterRecoverItem {
    name: string;
    hookName: string;
    result: string;
    msg: string;
}

File: pkg/service/dto/cluster.go
package dto

import "github.com/KubeOperator/KubeOperator/pkg/model"

type Cluster struct {
	model.Cluster
	NodeSize int    `json:"nodeSize"`
	Status   string `json:"status"`
}

type ClusterPage struct {
	Items []Cluster `json:"items"`
	Total int       `json:"total"`
}

type ClusterStatus struct {
	model.ClusterStatus
}

type ClusterSecret struct {
	model.ClusterSecret
}

type ClusterNode struct {
	model.ClusterNode
}

type ClusterMonitor struct {
	model.ClusterMonitor
}

type ClusterSpec struct {
	model.ClusterSpec
}

type NodeCreate struct {
	HostName string `json:"hostName"`
	Role     string `json:"role"`
}

type ClusterCreate struct {
	Name                 string       `json:"name" binding:"required"`
	Version              string       `json:"version" binding:"required"`
	NetworkType          string       `json:"networkType"`
	RuntimeType          string       `json:"runtimeType"`
	DockerStorageDIr     string       `json:"dockerStorageDIr"`
	ContainerdStorageDIr string       `json:"containerdStorageDIr"`
	AppDomain            string       `json:"appDomain"`
	ClusterCIDR          string       `json:"clusterCIDR"`
	ServiceCIDR          string       `json:"serviceCIDR"`
	Nodes                []NodeCreate `json:"nodes"`
}

type ClusterBatch struct {
	Items     []Cluster
	Operation string
}

type Endpoint struct {
	Address string
	Port    int
}
